src/test/java/org/sasanlabs/service/vulnerability/commandInjection/CommandInjectionTest.java

```diff
@@ -0,0 +1,125 @@
+        assertThat(responseEntity.getBody().getIsValid()).isEqualTo(true);
+
+        verify(commandInjection).getVulnerablePayloadLevel1(LOCALHOST);
+        verify(commandInjection).getResponseFromPingCommand(eq(LOCALHOST), eq(trueCondition));
+    }
+
+    @Test
+    public void getVulnerablePayloadLevel1_NoIp_ExpectNoResponseContent() throws IOException {
+        ResponseEntity<GenericVulnerabilityResponseBean<String>> responseEntity =
+                commandInjection.getVulnerablePayloadLevel1(Strings.EMPTY);
+
+        // Assert
+        assertThat(responseEntity.getBody().getContent()).isEqualTo(Strings.EMPTY);
+        assertThat(responseEntity.getStatusCode()).isEqualTo(HttpStatus.OK);
+        assertThat(responseEntity.getBody().getIsValid()).isEqualTo(true);
+
+        verify(commandInjection).getVulnerablePayloadLevel1(eq(Strings.EMPTY));
+        verify(commandInjection).getResponseFromPingCommand(eq(Strings.EMPTY), eq(falseCondition));
+    }
+
+    @Test
+    public void getVulnerablePayloadLevel2_Localhost_ExpectPingSuccess()
+            throws ServiceApplicationException, IOException {
+        RequestEntity entity = new RequestEntity(HttpMethod.GET, URI.create(LOCALHOST));
+        ResponseEntity<GenericVulnerabilityResponseBean<String>> responseEntity =
+                commandInjection.getVulnerablePayloadLevel2(LOCALHOST, entity);
+
+        assertThat(responseEntity.getBody().getContent()).isEqualTo(successReturnStr);
+        assertThat(responseEntity.getStatusCode()).isEqualTo(HttpStatus.OK);
+        assertThat(responseEntity.getBody().getIsValid()).isEqualTo(true);
+
+        verify(commandInjection).getVulnerablePayloadLevel2(eq(LOCALHOST), eq(entity));
+        verify(commandInjection).getResponseFromPingCommand(eq(LOCALHOST), eq(trueCondition));
+    }
+
+    @Test
+    public void getVulnerablePayloadLevel2_NoIp_ExpectNoResponseContent()
+            throws ServiceApplicationException, IOException {
+        RequestEntity entity = new RequestEntity(HttpMethod.GET, URI.create(Strings.EMPTY));
+        ResponseEntity<GenericVulnerabilityResponseBean<String>> responseEntity =
+                commandInjection.getVulnerablePayloadLevel2(Strings.EMPTY, entity);
+
+        // Assert
+        assertThat(responseEntity.getBody().getContent()).isEqualTo(Strings.EMPTY);
+        assertThat(responseEntity.getStatusCode()).isEqualTo(HttpStatus.OK);
+        assertThat(responseEntity.getBody().getIsValid()).isEqualTo(true);
+
+        verify(commandInjection).getVulnerablePayloadLevel2(eq(Strings.EMPTY), eq(entity));
+        verify(commandInjection).getResponseFromPingCommand(eq(Strings.EMPTY), eq(falseCondition));
+    }
+
+    @Test
+    public void getVulnerablePayloadLevel2_LocalhostAndSEMICOLON_ExpectNoResponseContent()
+            throws ServiceApplicationException, IOException {
+        RequestEntity entity =
+                new RequestEntity(HttpMethod.GET, URI.create("localhost;localhost2"));
```

Comment:
Please create a proper URI. for this you need to run this vulnerability and check what is being passed so that we can have a proper setup, similar to what we have when application runs.

Reply:
please start the vulnerableApp you will get the UI, go to command injection vulnerability and then give some input to input box, there will be one url fired to the backend, use that url as that is a correct url for command injection.